### ISSUE TITLE
Maintainance status

### ISSUE THREAD

**Issue by bebehei** [prosody/prosody-docker#44]
Hi all,

I've looked at the current prosody docker image. The current one looks rather unsatisfying.

- It builds on Ubuntu **14.04**.
- It requires always a correct precompiled deb package.
- It's way bigger than building it on Alpine (uncompressed: 200MB on Ubuntu vs <10MB on Alpine[0]).
- The dockerhub image doesn't build from github source.
- Some of the dependencies are outdated

Would you appreciate a PR to make this container build from source and use an alpine image?

Greets,
Benedikt

[0] I'm currently bootstrapping it from hg source on alpine. I have to write some additional scripts like a better entrypoint. But the raw build only requires 7.5MB.

**Comment by mwild1** [comment]
Hi Benedikt,

Thanks for the feedback.

>   It builds on Ubuntu 14.04.

I'm curious - do you know of any specific problems coming from this right now?

>    It requires always a correct precompiled deb package.

This is because the primary purpose of this repo is to provide docker images of our automated nightly and release builds (see .https://prosody.im/nightly/ ). Using the package simplifies a lot of things, because it already takes care of a default configuration for that distribution, creating prosody users, etc. - meaning you get the same kind of setup whether you use Docker or not.

It also means we can use the Docker images to easily verify that the packages are working correctly. Basically, everything that comes from more code re-use.

>    It's way bigger than building it on Alpine (uncompressed: 200MB on Ubuntu vs <10MB on Alpine[0]).

If image size is a concern, there is already an 'alpine' branch in this repository. Please check it out, and feedback/PRs gladly received.

>    The dockerhub image doesn't build from github source.

We don't host our source on Github, and since we already have nightly builds it's much easier for us to know that "nightly build 401" is the same code regardless of whether someone is using Docker, deb package, or compiled from source.

>   Some of the dependencies are outdated

I'll take a look (likely this is a good reason to update the distribution version!).

---

So, in summary... yes, it's maintained, and improvements and feedback are welcome. Just be aware that we're using it in a specific way. I'm not opposed to making some changes to our build server, e.g. perhaps the benefits of building from source on top of alpine outweigh the benefits of only maintaining default configuration and environment setup in one place (i.e. the deb package).

**Comment by bebehei** [comment]
@mwild1 Oh well, based on your fast response, I'll withdraw to call this unmaintained. Also thanks for elaborating your point of view.

> I'll take a look (likely this is a good reason to update the distribution version!).

Oh, well there is this #18. I consider rather critical, that such an easy fix isn't deployed yet to a problem, which got reported >2 years ago.

>  there is already an 'alpine' branch in this repository

Oh. I've looked at it. It still copies the debian package file 😂 This gives me the impression, that this hasn't received and probably will not receive enough work to be useful. Also `prosody/prosody-alpine` is not updated for the last two years.

> > The dockerhub image doesn't build from github source.

Sorry, I didn't express myself good enough, there.

Idealistically, you can state in a Dockerfile ARG any branch/tag (or also a raw commit id/revision), which then builds an image with this exact revision (not required to build from source). Better of course would be the existence of the Dockerfile in the actual repository and then enable CI(/CD) for Docker.

The actual problem is this here:

> The dockerhub image doesn't build from github source.

Currently, neither I nor dockerhub can actually build this image out of the GitHub repository. If you make this image download its package from the prosody nightly build website and do it inside the docker container, others can verify the image (reproducibility). This will also make contributions to this docker container easier, as I don't have to manually get a prosody package.

Also, this will be possible to do in respect to your actual purpose of this repository.

**Comment by Zash** [comment]
Any reason for this to have been reopened?

**Comment by bebehei** [comment]
It seems that I accidentally closed this issue while submitting my comment. As the initial points do not seem to apply anymore, I'm going to close this issue.
